Update by ZIP version

MaxTNS opened the pull request.

MaxTNS committed:
Update main.yml

MaxTNS committed:
Update main.yml

MaxTNS committed:
Update youtrack.service.j2

oukooveu commented on tasks/main.yml:
```diff
@@ -38,21 +38,35 @@
-    checksum: "sha256:https://download.jetbrains.com/charisma/youtrack-{{ youtrack_version }}.jar.sha256"
+    url: "https://download.jetbrains.com/charisma/youtrack-{{ youtrack_version }}.zip"
+    dest: "{{ youtrack_releases_dir }}/youtrack-{{ youtrack_version }}.zip"
+#    checksum: "sha256:https://download.jetbrains.com/charisma/youtrack-{{ youtrack_version }}.zip.sha256"
```
why this was commented out?

MaxTNS replied:
If will be honest, i don't know why but i'm get error on this step.

oukooveu replied:
Is this reproducible? Could you please provide error message?

MaxTNS replied:
Yeap. this is reproducible. 

Text of error message: 

TASK [ary2 : download youtrack zip file] *********************************************************************************************************************************************************************
fatal: [u2]: FAILED! => {"changed": false, "msg": "Unable to find a checksum for file 'youtrack-2023.1.17582.zip' in 'https://download.jetbrains.com/charisma/youtrack-2023.1.17582.zip.sha256'"}

Meanwhile i check link from official site: https://download.jetbrains.com/charisma/youtrack-2023.1.17582.zip.sha256

Both link indentical.

oukooveu replied:
Both links work for me:
![image](https://github.com/oukooveu/ansible-role-youtrack/assets/2072275/18ca4a89-f48b-4124-81e1-e18523db9869)

MaxTNS replied:
For me too, but while ansible trying to complete this step that's crushing every times.

oukooveu replied:
The playbook below works as expected, please check your setup:
```- name: test download
  hosts: localhost
  gather_facts: false
  vars:
    youtrack_version: 2023.1.17582
    youtrack_releases_dir: /tmp
  tasks:
    - name: download youtrack zip file
      ansible.builtin.get_url:
        url: "https://download.jetbrains.com/charisma/youtrack-{{ youtrack_version }}.zip"
        dest: "{{ youtrack_releases_dir }}/youtrack-{{ youtrack_version }}.zip"
        checksum: "sha256:https://download.jetbrains.com/charisma/youtrack-{{ youtrack_version }}.zip.sha256"
```

oukooveu commented on tasks/main.yml:
```diff
@@ -38,21 +38,35 @@
+    remote_src: yes
+
+- name: Mkdir bin
+  shell: mkdir -p /home/youtrack/bin
```
`ansible.builtin.file` but what's the reason to create `bin` directory and not to use script from youtrack distribution?

MaxTNS replied:
I'm just tryna to follow to logic of previous version with simbolic link.

oukooveu commented on templates/youtrack.service.j2:
```diff
@@ -1,14 +1,15 @@
+
 
 [Install]
-WantedBy=default.target
```
no changes

oukooveu reviewed (commented):
i'm going to fix tests soon

MaxTNS reviewed (commented):
Except other edited lines was add new step for showing token after start the service. 'cause previously we was in need to read ".txt" file for grub our new token for continue our update. And now we can copy token from output in shell by ansible.

+  -name: Read token
+  -name: Show token

oukooveu commented:
tests have been fixed, please rebase

MaxTNS committed:
Update youtrack.service.j2

Change path to script directory

MaxTNS committed:
Update main.yml

- Was deleted step with creation of symbolic link
- Added step with setting ownership for "youtrack_release_dir"

MaxTNS committed:
Merge branch 'oukooveu:main' into second

MaxTNS committed:
Update main.yml

MaxTNS committed:
Update youtrack.service.j2

MaxTNS committed:
Update main.yml

MaxTNS committed:
Update youtrack.service.j2

MaxTNS committed:
Update main.yml

MaxTNS committed:
Update test_main.py

MaxTNS committed:
Update test_main.py

MaxTNS reviewed (commented):
Test was modificated for check create symlink to directory.

MaxTNS reviewed (commented):
Was edited path for symlink

MaxTNS committed:
Added checks to main.yml

Was added two checks:
- Checking previously downloaded zip file
- Checking previously unzipped file

oukooveu commented on tasks/main.yml:
```diff
@@ -38,23 +40,61 @@
     owner: "{{ youtrack_user }}"
     group: "{{ youtrack_group }}"
     mode: 0640
+  when: not youtrack_file_exists.stat.exists
```
unnecessary because of https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html#parameter-checksum

oukooveu commented on tasks/main.yml:
```diff
@@ -38,23 +40,61 @@
+    path: "{{ youtrack_releases_dir }}/youtrack-{{ youtrack_version }}.zip"
+  register: youtrack_file_exists
+
+- name: debug youtrack file status
```
please remove

oukooveu commented on tasks/main.yml:
```diff
@@ -38,23 +40,61 @@
 
-- name: download youtrack jar file
+
+- name: check if youtrack zip file exists
```
isn't required, see comment below

oukooveu commented on tasks/main.yml:
```diff
@@ -73,4 +113,12 @@
+  shell: cat /home/youtrack/releases/youtrack-{{ youtrack_version }}/conf/internal/services/configurationWizard/wizard_token.txt
+  register: token
+
+- name: show token
```
insecure, please remove

oukooveu commented on tasks/main.yml:
```diff
@@ -38,23 +40,61 @@
+    src: "{{ youtrack_releases_dir }}/youtrack-{{ youtrack_version }}.zip"
+    dest: "{{ youtrack_releases_dir }}/"
+    remote_src: yes
+  when: zip_file.stat.exists == false
```
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html#parameter-creates
